contracts/contracts/staking/SingleAssetStaking.sol

```diff
@@ -169,6 +170,10 @@ contract SingleAssetStaking is Initializable, Governable {
 
         require(i < MAX_STAKES, "Max stakes");
 
+        if (i == 0 || stakes[i - 1].paid) {
```

Comment:
Super nit: to be explicit, add a comment explaining that if it's a new staker or a staker that had exited all their stakes, we increment the counter.